main.py

```diff
@@ -377,28 +402,32 @@ def main():
         choices=["DEBUG", "INFO", "WARN", "ERROR"],
         help="Minimum loglevel for logging",
     )
+    parser.add_argument(
+        "--print_report",
+        dest="print_report",
+        action="store_true",
+        help="Print a detailed report of all file processed.",
+    )
 
-    global global_arguments
-    global_arguments = parser.parse_args()
+    args = Arguments(**vars(parser.parse_args()))
```

Comment:
@ritwik-g The parsed arguments returned by `argparse.ArgumentParser` itself might be sufficient for use everywhere.